src/main/java/com/ohmyclass/api/components/subject/repository/ISubjectRepository.java

```diff
@@ -1,5 +1,16 @@
 package com.ohmyclass.api.components.subject.repository;
 
-public interface ISubjectRepository {
+import java.util.Optional;
 
+import org.springframework.data.repository.CrudRepository;
+
+import com.ohmyclass.api.components.group.entity.Group;
+import com.ohmyclass.api.components.subject.entity.Subject;
+
+public interface ISubjectRepository extends CrudRepository<Subject, Long> {
+
+    Optional<Subject> findSubjectById(Long id);
+
+    Optional<Subject> getSubjectByGroup(Group group);
```

Comment:
The "standard" you've set for queries is to find entities by ...-ids.
Also: I'm not sure by the "get".
Change to findSubjectByGroupId(Long groupId)